src/_gettsim/taxes/lohnst.py

```diff
@@ -0,0 +1,401 @@
+      See :func:`ges_pflegev_zusatz_kinderlos`
+    wohnort_ost:
+      See basic input variable :ref:`wohnort_ost <wohnort_ost>`.
+    ges_krankenv_zusatzbeitrag
+        See :func:ges_krankenv_zusatzbeitrag`.
+    eink_st_abzuege_params:
+      See params documentation :ref:`eink_st_abzuege_params`
+    soz_vers_beitr_params:
+        See params documentation :ref:`soz_vers_beitr_params`
+
+
+    Returns
+    -------
+    Individual Vorsorgepauschale on annual basis
+
+    """
+
+    # 1. Rentenversicherungsbeiträge, §39b (2) Nr. 3a EStG.
+    if wohnort_ost:
+        bruttolohn_rente = min(
+            12 * bruttolohn_m,
+            12 * soz_vers_beitr_params["beitr_bemess_grenze_m"]["ges_rentenv"]["ost"],
+        )
+    else:
+        bruttolohn_rente = min(
+            12 * bruttolohn_m,
+            12 * soz_vers_beitr_params["beitr_bemess_grenze_m"]["ges_rentenv"]["west"],
+        )
+
+    vorsorg_rv = (
+        bruttolohn_rente
+        * soz_vers_beitr_params["beitr_satz"]["ges_rentenv"]
+        * eink_st_abzuege_params["vorsorgepauschale_rv_anteil"]
+    )
+
+    # 2. Krankenversicherungsbeiträge, §39b (2) Nr. 3b EStG.
+    # For health care deductions, there are two ways to calculate.
+    # a) at least 12% of earnings of earnings can be deducted,
+    #    but only up to a certain threshold
+
+    bruttolohn_kv = min(
+        12 * bruttolohn_m,
+        12 * soz_vers_beitr_params["beitr_bemess_grenze_m"]["ges_krankenv"]["ost"],
+    )
+    vorsorg_kv_option_a_basis = (
+        eink_st_abzuege_params["vorsorgepauschale_mindestanteil"] * bruttolohn_kv
+    )
+
+    if steuerklasse == 3:
+        vorsorg_kv_option_a_max = eink_st_abzuege_params["vorsorgepauschale_kv_max"][
+            "stkl3"
+        ]
+    else:
+        vorsorg_kv_option_a_max = eink_st_abzuege_params["vorsorgepauschale_kv_max"][
+            "stkl_nicht3"
+        ]
+
+    vorsorg_kv_option_a = min(vorsorg_kv_option_a_max, vorsorg_kv_option_a_basis)
+
+    # b) Take the actual contributions (usually the better option),
+    #   but apply the reduced rate!
+
+    if ges_pflegev_zusatz_kinderlos:
+        beitr_satz_pflegev = (
+            soz_vers_beitr_params["beitr_satz"]["ges_pflegev"]["standard"]
+            + soz_vers_beitr_params["beitr_satz"]["ges_pflegev"]["zusatz_kinderlos"]
+        )
+    else:
+        beitr_satz_pflegev = soz_vers_beitr_params["beitr_satz"]["ges_pflegev"][
+            "standard"
+        ]
+
+    vorsorg_kv_option_b = bruttolohn_kv * (
+        soz_vers_beitr_params["beitr_satz"]["ges_krankenv"]["ermäßigt"] / 2
+        + ges_krankenv_zusatzbeitrag / 2 / 100
+        + beitr_satz_pflegev
+    )
+
+    # add both RV and KV deductions. For KV, take the larger amount.
+    out = vorsorg_rv + max(vorsorg_kv_option_a, vorsorg_kv_option_b)
+    return out
+
+
+@add_rounding_spec(params_key="lohnst")
+def vorsorgepauschale_ab_2005_bis_2009() -> float:
+    """vorsorg_rv and vorsorg_kv_option_a are identical to after 2010."""
```

Comment:
This docstring refers to parameters not being used here and to a different time period than suggested by the function. Please fix; maybe get rid of this altogether? What is the point of returning 0, sounds like another interface should see adjustment ?

Reply:
Thanks. I guess this dosctring belongs to the (non-existent) function for up to 2004. We currently have `vorsorgepauschale` properly since 2010 only. @JakobWegmann: do we want to model the 2005 rule in this PR as well?

Reply:
Probably not, in that case, however, just open an issue for it and delete the code. Misleading docstrings are worse than no docs. :smile:

Reply:
Something for the future :)